File: EzyBill.BLL/Models/DataTranserObjects/InvoiceDto.cs
using EzyBill.BLL.Enums;
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EzyBill.BLL.Models.DataTranserObjects
{
    public class InvoiceDto
    {
        public Guid Id { get; set; }
        public string CustomerName { get; set; } = string.Empty;
        public string BillingAddressLine1 { get; set; } = string.Empty;
        public string BillingAddressLine2 { get; set; } = string.Empty;
        public IEnumerable<ProductLineItemDto> InvoicedProducts { get; set; } = Enumerable.Empty<ProductLineItemDto>();
        public PaymentStatus PaymentStatus { get; set; } = PaymentStatus.Unpaid;
        public double AmountDue { get; set; }
    }
}

File: EzyBill.Web/src/app/shared/interfaces/invoice-dto.ts
import { IProductLineItemDto } from "./product-line-item-dto";

export enum PaymentStatus
    {
        Unpaid = 0,
        Paid = 1
    }
export interface IInvoiceDto {
        id: string;
        customerName: string;
        billingAddressLine1: string;
        billingAddressLine2: string;
        InvoicedProducts: Array<IProductLineItemDto>;
        PaymentStatus: PaymentStatus;
        amountDue: number;
}
